uint8 id
int16 x
int16 y
int16 z